--- a/src/App/Application.cpp
+++ b/src/App/Application.cpp
@@ -1732,7 +1732,9 @@ static void freecadNewHandler ()
 #endif
 
 #if defined(FC_OS_LINUX)
+#ifdef __GLIBC__
 #include <execinfo.h>
+#endif
 #include <dlfcn.h>
 #include <cxxabi.h>
 
@@ -1749,6 +1751,7 @@ static void freecadNewHandler ()
 void printBacktrace(size_t skip=0)
 {
 #if defined HAVE_BACKTRACE_SYMBOLS
+#ifdef __GLIBC__
     void *callstack[128];
     size_t nMaxFrames = sizeof(callstack) / sizeof(callstack[0]);
     size_t nFrames = backtrace(callstack, nMaxFrames);
