--- a/src/3rdParty/salomesmesh/src/DriverSTL/Basics_Utils.cpp
+++ b/src/3rdParty/salomesmesh/src/DriverSTL/Basics_Utils.cpp
@@ -29,9 +29,10 @@
 #ifndef WIN32
 #include <unistd.h>
 #include <sys/stat.h>
+#ifdef __GLIBC__
 #include <execinfo.h>
 #endif
-
+#endif
 
 namespace Kernel_Utils
 {
@@ -109,6 +110,7 @@ namespace Kernel_Utils
 #ifndef WIN32
   void print_traceback()
   {
+#ifdef __GLIBC__
     void *array[50];
     size_t size;
     char **strings;
@@ -123,6 +125,9 @@ namespace Kernel_Utils
       }
 
     free (strings);
+#else
+   std::cerr << "sorry, no backtrace on musl libc";
+#endif
   }
 #else
   #if (_MSC_VER >= 1400) // Visual Studio 2005
